realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/SyncSessionTests.java

```diff
@@ -455,4 +462,121 @@ public void run() {
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(9);
+        objectA.setString("partial");
+
+        realm.createObject(PartialSyncObjectB.class).setNumber(0);
+        realm.createObject(PartialSyncObjectB.class).setNumber(1);
+        realm.createObject(PartialSyncObjectB.class).setNumber(2);
+        realm.createObject(PartialSyncObjectB.class).setNumber(3);
+        realm.createObject(PartialSyncObjectB.class).setNumber(4);
+        realm.createObject(PartialSyncObjectB.class).setNumber(5);
+        realm.createObject(PartialSyncObjectB.class).setNumber(6);
+        realm.createObject(PartialSyncObjectB.class).setNumber(7);
+        realm.createObject(PartialSyncObjectB.class).setNumber(8);
+        realm.createObject(PartialSyncObjectB.class).setNumber(9);
+        realm.commitTransaction();
+
+        SyncManager.getSession(adminConfig).uploadAllLocalChanges();
+        realm.close();
+
+        final CountDownLatch latch = new CountDownLatch(2);
+
+        HandlerThread handlerThread = new HandlerThread("background");
+        handlerThread.start();
+        Handler handler = new Handler(handlerThread.getLooper());
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                final Realm partialSyncRealm = Realm.getInstance(partialSyncConfig);
+                assertTrue(partialSyncRealm.isEmpty());
+
+                partialSyncRealm.subscribeToObjects(PartialSyncObjectA.class, "number > 5", new Realm.PartialSyncCallback<PartialSyncObjectA>() {
+
+                    @Override
+                    public void onSuccess(RealmResults<PartialSyncObjectA> results) {
+                        assertEquals(4, results.size());
+                        for (PartialSyncObjectA object : results) {
+                            assertThat(object.getNumber(), greaterThan(5));
+                            assertEquals("partial", object.getString());
+                        }
+                        // make sure the Realm contains only PartialSyncObjectA
+                        assertEquals(0, partialSyncRealm.where(PartialSyncObjectB.class).count());
+                        latch.countDown();
+                    }
+
+                    @Override
+                    public void onError(RealmException error) {
+                        fail(error.getMessage());
+                    }
+                });
+
+                // Invalid query
+                partialSyncRealm.subscribeToObjects(PartialSyncObjectA.class, "invalid_property > 5", new Realm.PartialSyncCallback<PartialSyncObjectA>() {
+
+                    @Override
+                    public void onSuccess(RealmResults<PartialSyncObjectA> results) {
+                        fail("Invalid query should not succeed");
+                    }
+
+                    @Override
+                    public void onError(RealmException error) {
+                        assertNotNull(error);
```

Comment:
What kind of error is thrown here?

Reply:
This will wrap the error message we get from the Sync client as a `RealmException` https://github.com/realm/realm-object-store/pull/547/files#diff-854aeb40d09f785f4384eb120ed94eadR106